### ISSUE TITLE
[Bug]: Podcast RSS Download with long titles

### ISSUE THREAD

**Issue by ofifoto** [advplyr/audiobookshelf#663]
### Describe the issue

Hi! One podcast I subscribe to has super long titles sometimes (and sometimes with a newline), which causes audiobookshelf to crash/fail to download episodes. It'd be great if it could catch this somehow trim the title to be within limits.

Sadly I can't convince the publisher to shorten or fix the titles

### Steps to reproduce the issue

1. Click "Find Episodes"
2. Select all
3. Start downloading

here's the logs with a (censored, but same length) path that triggered the I/O error. (storage backend is rclone, unsure if other systems have similar limits)

```
/server/njodb/index.js:103                                                                                                              
        throw error;                                                                                                                                                                                                                                                             
        ^                                                                                                                               
                                                                                                                                                                                                                                                                                 
Error: EIO: i/o error, open '/podcasts/xxxxxxxxxxxxx/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
xxxxxxxxxxxxxxxxxxxxxx.mp3'                                                                                              
Emitted 'error' event on WriteStream instance at:                                                                                       
    at emitErrorNT (node:internal/streams/destroy:157:8)                                                                                
    at emitErrorCloseNT (node:internal/streams/destroy:122:3)                                                                                                                                                                                                                    
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {                                                                                                                                                                                                     
  errno: -5,                                                                                                                                                                                                                                                                     
  code: 'EIO',                                                                                                                          
  syscall: 'open',
  path: '/podcasts/xxxxxxxxxxxxx/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
xxxxxxxxxxxxxxxxxxxxxx.mp3'
}
```

### Audiobookshelf version

v2.0.18

### How are you running audiobookshelf?

Docker

**Comment by advplyr** [comment]
That's strange to put newlines in an episode title, but either way we should be sanitizing the filename so thanks for catching that.

**Comment by advplyr** [comment]
Actually we are already sanitizing that filename, can you give a little more of the log since I'm not sure what is triggering that error. Is the episode getting saved at all to the file system?

**Comment by ofifoto** [comment]
Will do! I’m not at my laptop at the moment but maybe it has to do with file length? I tried manually creating a file with the same name with `touch` sans new line and got an i/o error but when i hacked off a big chunk of the name touch succeeded

**Comment by ofifoto** [comment]
Ah! Thanks for that. Will test and get back to you if it persists :)

**Comment by advplyr** [comment]
Fixed in [v2.0.19](https://github.com/advplyr/audiobookshelf/releases/tag/v2.0.19)
